ds/ds.go

```diff
@@ -124,13 +125,35 @@ func Parse(uri string) (dsQuery, error) {
 		return result, fmt.Errorf("Not an dns-sd URI")
 	}
 
-	// following dns-sd URI conventions from CUPS
-	if u.Host != "" {
-		result.Domain = u.Host
-	}
-	if u.Path != "" {
-		// remove prefix slash
-		result.Type = u.Path[1:]
+	// following dns-sd URI conventions from CUPS dnssd://instance._type._proto.domain/?query
+	if u.Hostname() != "" {
+		parts := strings.Split(u.Hostname(), ".")
+		found := false
+		for p, v := range parts {
+			if v[0] == '_' {
+				if p > 0 {
+					result.Instance = strings.Join(parts[0:p], ".")
+				}
+				if len(parts) > p+1 {
```

Comment:
Some sort of comment about what is going on here would be helpful for future readers maybe.